File: Front-end/app/lib/models/Plan.dart
class Plan {
  final String? uuid;
  final String name;
  final int? price;
  final int? durationDays;
  final String? description;

  Plan({
    this.uuid,
    required this.name,
    this.price,
    this.durationDays,
    this.description,
  });

  factory Plan.fromJson(Map<String, dynamic> json) {
    return Plan(
      uuid: json['uuid'],
      name: json['name'] ?? '',
      price: json['price'] != null ? int.parse(json['price'].toString()) : null,
      durationDays: json['durationDays'] != null
          ? int.parse(json['durationDays'].toString())
          : null,
      description: json['description'],
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'uuid': uuid,
      'name': name,
      'price': price,
      'durationDays': durationDays,
      'description': description,
    };
  }
}

File: Back-end/gym-server/src/main/java/com/lht/pojo/Plan.java
package com.lht.pojo;

import jakarta.persistence.*;
import lombok.*;

import java.util.UUID;

@Entity
@Table(name = "plan")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Plan {

    @Id
    @Column(columnDefinition = "uuid", updatable = false, nullable = false)
    private UUID uuid;

    @Column(length = 100)
    private String name;

    @Column(nullable = false)
    private Integer price;

    @Column(name = "duration_days", nullable = false)
    private Integer durationDays;

    @Column(columnDefinition = "TEXT")
    private String description;
}